mailpoet/assets/js/src/email-editor/engine/components/styles-sidebar/preview.tsx

```diff
@@ -38,181 +41,191 @@ const secondFrame = {
 const normalizedHeight = 152;
 const normalizedColorSwatchSize = 32;
 
+type Props = {
+  label?: string;
+  isFocused?: boolean;
+  withHoverView?: boolean;
+};
+
 /**
  * Component to render the styles preview based on the component from the site editor:
  * https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/preview.js
  */
-export function Preview(): JSX.Element {
-  const style = {
-    backgroundColor: '#f3f3f3',
-    headingFontFamily: 'Arial',
-    headingColor: '#000000',
-    headingFontWeight: 'normal',
-    paletteColors: [
-      {
-        name: 'Sample Background',
-        slug: 'Sample-background',
-        color: '#f9f8f3',
-      },
-    ],
-    highlightedColors: [
-      {
-        name: 'Sample primary',
-        slug: 'sample-primary',
-        color: '#e5e2d3',
-      },
-      {
-        name: 'Sample Secondary',
-        slug: 'sample-secondary',
-        color: '#111111',
-      },
-    ],
-  };
+export function Preview({
+  label,
+  isFocused,
+  withHoverView,
+}: Props): JSX.Element {
+  const styles = useSelect((select) => select(storeName).getStyles());
 
-  const isFocused = false;
-  const isHovered = false;
-  const disableMotion = false;
-  const label = 'Some Label';
-  const ratio = 1;
+  const backgroundColor = styles.colors.background;
+  const headingFontFamily = styles.elements.h1.fontFamily;
+  const headingColor = styles.elements.h1.color;
+  const headingFontWeight = styles.elements.h1.fontWeight;
+  const paletteColors = [
+    {
+      name: 'Sample primary',
+      slug: 'sample-primary',
+      color: '#a5a2a3',
+    },
+    {
+      name: 'Sample Secondary',
+      slug: 'sample-secondary',
+      color: '#e5e2d3',
+    },
+    {
+      name: 'Sample Background',
+      slug: 'Sample-background',
+      color: '#a8c7a9',
+    },
+  ];
+  // https://github.com/WordPress/gutenberg/blob/7fa03fafeb421ab4c3604564211ce6007cc38e84/packages/edit-site/src/components/global-styles/hooks.js#L68-L73
```

Comment:
Where will we get these colors from?

I checked the link and if I understand correctly the primary is text color, the secondary is heading color and the background is document background. Am I right?

Shouldn't we set those in theme.json and apply them here as defaults? Later if a user sets their own, we should merge user-defined data with our theme.json data, and user's would win.